### DISCUSSION TITLE
automatically add curent repo path to turbo gen

### DISCUSSION THREAD

**Discussion by bogris** [vercel/turborepo#8663] [Ideas]
### Goals

enable easy usage of global generators in different packages

### Non-goals

provide more context to plop

### Background

Let't assume we are trying to create a global generator that will be used in multiple packages. EX: react component generator shared across multiple libraries. 

We can put the generator in the root of the project. 
As of now, the turbo gen takes the base path of the generator, so the generated files will be relative to the root of the project. 

If we need to generate a package specific component, we need to manually supply the package name.

I feel that a better DX would be to run the generator in the root of a package folder so we can specify this way the package like this. But there is no way to know where the command is run. 

### Proposal

 turbo gen can pass the current directory where the command is executed in and ENV variable to plop. 

this would also provide more context that can be used by others too hook in.

**Comment by anthonyshew** [comment]
Happy to take a PR for this!
